src/pipecat/services/sarvam/tts.py

```diff
@@ -240,3 +248,327 @@ async def run_tts(self, text: str) -> AsyncGenerator[Frame, None]:
+            speech_sample_rate: Sampling frequency for output audio (default 24000 Hz).
+            enable_preprocessing: If true, applies text normalization/preprocessing.
+            min_buffer_size: Minimum number of tokens to buffer before synthesis (default 50).
+            max_chunk_length: Maximum number of tokens in one chunk (default 200).
+            output_audio_codec: Audio codec of output frames (default “linear16”).
+            output_audio_bitrate: Bitrate for output audio format (default “128k”).
+            **kwargs: Optional keyword arguments forwarded to WebsocketTTSService (such as
+                `push_stop_frames`, `sample_rate`, task manager parameters, event hooks, etc.)
+                to customize transport behavior or enable metrics support.
+
+        This method sets up the internal TTS configuration mapping, constructs the WebSocket
+        URL based on the chosen model, and initializes state flags before connecting.
+        """
+        # Initialize parent class first
+        super().__init__(
+            push_stop_frames=True,
+            sample_rate=speech_sample_rate,
+            **kwargs,
+        )
+
+        # WebSocket endpoint URL
+        self._websocket_url = f"wss://api.sarvam.ai/text-to-speech/ws?model={model}"
+        self._api_key = api_key
+        self.set_model_name(model)
+        self.set_voice(speaker)
+        # Configuration parameters
+        self._config = {
+            "target_language_code": target_language_code,
+            "pitch": pitch,
+            "pace": pace,
+            "speaker": speaker,
+            "loudness": loudness,
+            "speech_sample_rate": speech_sample_rate,
+            "enable_preprocessing": enable_preprocessing,
+            "min_buffer_size": min_buffer_size,
+            "max_chunk_length": max_chunk_length,
+            "output_audio_codec": output_audio_codec,
+            "output_audio_bitrate": output_audio_bitrate,
+        }
+        # self._websocket = None
+        self._started = False
+        self._cumulative_time = 0
+
+        self._receive_task = None
+        self._keepalive_task = None
+
+    def can_generate_metrics(self) -> bool:
+        """Check if this service can generate processing metrics.
+
+        Returns:
+            True, as Sarvam service supports metrics generation.
+        """
+        return True
+
+    def language_to_service_language(self, language: Language) -> Optional[str]:
+        """Convert a Language enum to Sarvam AI language format.
+
+        Args:
+            language: The language to convert.
+
+        Returns:
+            The Sarvam AI-specific language code, or None if not supported.
+        """
+        return language_to_sarvam_language(language)
+
+    async def start(self, frame: StartFrame):
+        """Start the Sarvam TTS service.
+
+        Args:
+            frame: The start frame containing initialization parameters.
+        """
+        await super().start(frame)
```

Comment:
Assuming the naming changes I'm recommending, we initialize the sample_rate here.

```suggestion
        await super().start(frame)
        self._settings["sample_rate"] = self.sample_rate
```

This sets the value from the TTSService base class, which is handling the StartFrame and exposing a `sample_rate` property.

Reply:
Addressed in a PR